firebase-firestore/src/test/java/com/google/firebase/firestore/local/RemoteDocumentCacheTestCase.java

```diff
@@ -241,12 +245,39 @@ public void testGetAllFromUsesReadTimeNotUpdateTime() {
     addTestDocumentAtPath("b/old", /* updateTime= */ 1, /* readTime= */ 2);
     addTestDocumentAtPath("b/new", /* updateTime= */ 2, /* readTime= */ 1);
 
-    ResourcePath collection = path("b");
     Map<DocumentKey, MutableDocument> results =
-        remoteDocumentCache.getAll(collection, IndexOffset.createSuccessor(version(1), -1));
+        remoteDocumentCache.getDocumentsMatchingQuery(
+            query("b"), IndexOffset.createSuccessor(version(1), -1), new HashSet<DocumentKey>());
     assertThat(results.values()).containsExactly(doc("b/old", 1, DOC_DATA));
   }
 
+  @Test
+  public void testGetMatchingDocsAppliesQueryCheck() {
+    addTestDocumentAtPath("a/1", 1, 1, map("matches", true));
+    addTestDocumentAtPath("a/2", 1, 2, map("matches", true));
+    addTestDocumentAtPath("a/3", 1, 3, map("matches", false));
+
+    Map<DocumentKey, MutableDocument> results =
+        remoteDocumentCache.getDocumentsMatchingQuery(
+            query("a").filter(filter("matches", "==", true)),
+            IndexOffset.createSuccessor(version(1), -1),
+            new HashSet<DocumentKey>());
+    assertThat(results.values()).containsExactly(doc("a/2", 1, map("matches", true)));
+  }
+
+  @Test
+  public void testGetMatchingDocsRespectsMutatedDocs() {
+    addTestDocumentAtPath("a/1", 1, 1, map("matches", true));
+    addTestDocumentAtPath("a/2", 1, 2, map("matches", false));
+
+    Map<DocumentKey, MutableDocument> results =
+        remoteDocumentCache.getDocumentsMatchingQuery(
+            query("a").filter(filter("matches", "==", true)),
+            IndexOffset.createSuccessor(version(1), -1),
+            new HashSet<DocumentKey>(Collections.singletonList(key("a/2"))));
+    assertThat(results.values()).containsExactly(doc("a/2", 1, map("matches", false)));
```

Comment:
want to make sure IIUC:
a/1 is excluded because we're looking for anything _after_ version 1.
a/2 should be excluded because it has matches==false, but because it's in the list of mutated documents, it is included.
correct?